react-emotion is missing react (peer)dependency

There is a direct dependency on react in your code but there is no reference to it in package.json

https://github.com/emotion-js/emotion/blob/d9a9e332dfbc13ea78325082393ccdd54d34afd6/packages/react-emotion/src/index.js#L2

https://github.com/emotion-js/emotion/blob/d9a9e332dfbc13ea78325082393ccdd54d34afd6/packages/react-emotion/package.json#L21-L27

This would normally not be an issue when using a package manager (NPM) that optimizes the resulting dependency tree since react will be placed at the root of the node_modules directory. Otherwise you'll get an error like me:

> Module not found: Error: Can't resolve 'react' in ...

--- a/packages/create-emotion-styled/package.json
+++ b/packages/create-emotion-styled/package.json
@@ -21,13 +21,13 @@
     "@emotion/is-prop-valid": "^0.6.1"
   },
   "peerDependencies": {
-    "prop-types": ">= 15"
+    "prop-types": "^15.6.1"
   },
   "devDependencies": {
     "@types/react": "16.0.16",
     "dtslint": "^0.3.0",
     "npm-run-all": "^4.0.2",
-    "prop-types": "^15.6.0",
+    "prop-types": "^15.6.1",
     "rimraf": "^2.6.1",
     "rollup": "^0.51.3"
   },

--- a/packages/create-emotion-styled/package.json
+++ b/packages/create-emotion-styled/package.json
@@ -21,7 +21,7 @@
     "@emotion/is-prop-valid": "^0.6.1"
   },
   "peerDependencies": {
-    "prop-types": "^15.6.1"
+    "prop-types": "15.x"
   },
   "devDependencies": {
     "@types/react": "16.0.16",

--- a/packages/emotion-theming/package.json
+++ b/packages/emotion-theming/package.json
@@ -47,7 +47,7 @@
     "hoist-non-react-statics": "^2.3.1"
   },
   "peerDependencies": {
-    "prop-types": "^15.6.1",
+    "prop-types": "15.x",
     "react": "15.x || 16.x"
   }
 }

--- a/packages/emotion-theming/package.json
+++ b/packages/emotion-theming/package.json
@@ -39,7 +39,7 @@
     "@types/react": "16.0.16",
     "cross-env": "^5.0.1",
     "dtslint": "^0.2.0",
-    "prop-types": "^15.5.8",
+    "prop-types": "^15.6.1",
     "rimraf": "^2.6.1",
     "rollup": "^0.51.3"
   },
@@ -47,7 +47,7 @@
     "hoist-non-react-statics": "^2.3.1"
   },
   "peerDependencies": {
-    "prop-types": ">= 15",
-    "react": ">= 15"
+    "prop-types": "^15.6.1",
+    "react": "15.x || 16.x"
   }
 }

--- a/packages/preact-emotion/package.json
+++ b/packages/preact-emotion/package.json
@@ -21,7 +21,7 @@
   },
   "peerDependencies": {
     "emotion": "^9.1.3",
-    "preact": ">= 8"
+    "preact": "8.x"
   },
   "devDependencies": {
     "emotion": "^9.1.3",

--- a/packages/preact-emotion/package.json
+++ b/packages/preact-emotion/package.json
@@ -20,7 +20,8 @@
     "create-emotion-styled": "^9.1.2"
   },
   "peerDependencies": {
-    "emotion": "^9.1.3"
+    "emotion": "^9.1.3",
+    "preact": ">= 8"
   },
   "devDependencies": {
     "emotion": "^9.1.3",

--- a/packages/react-emotion/package.json
+++ b/packages/react-emotion/package.json
@@ -23,7 +23,8 @@
     "create-emotion-styled": "^9.1.2"
   },
   "peerDependencies": {
-    "emotion": "^9.1.3"
+    "emotion": "^9.1.3",
+    "react": ">= 15"
   },
   "devDependencies": {
     "@types/react": "16.0.16",

--- a/packages/react-emotion/package.json
+++ b/packages/react-emotion/package.json
@@ -24,7 +24,7 @@
   },
   "peerDependencies": {
     "emotion": "^9.1.3",
-    "react": ">= 15"
+    "react": "15.x || 16.x"
   },
   "devDependencies": {
     "@types/react": "16.0.16",

--- a/packages/react-emotion/types/index.d.ts
+++ b/packages/react-emotion/types/index.d.ts
@@ -35,7 +35,7 @@ type ElementProps<Tag extends keyof JSX.IntrinsicElements> =
   & JSX.IntrinsicElements[Tag]
   & { innerRef?: JSX.IntrinsicElements[Tag]['ref'] };
 
-// tslint:disable:no-empty-interface
+// tslint:disable-next-line:no-empty-interface
 interface ComponentRef {}
 
 export interface StyledComponent<Props, Theme, IntrinsicProps>

